tools/REST.py

```diff
@@ -0,0 +1,106 @@
+import logging
+import os
+
+from fastapi import FastAPI
+from fastapi.middleware.cors import CORSMiddleware
+from gui import Job, JobFolder
+
+from utils import read_ctoaster_config
+
+app = FastAPI()
+
+# CORS configuration
+origins = [
+    "http://localhost:5001",  # React development server
+]
+
+app.add_middleware(
+    CORSMiddleware,
+    allow_origins=origins,
+    allow_credentials=True,
+    allow_methods=["*"],
+    allow_headers=["*"],
+)
+
+# Initialize the configuration
+read_ctoaster_config()
+
+
+@app.get("/jobs")
+def list_jobs():
+    try:
+        # Access the global variable ctoaster_jobs after it has been initialized
+        from utils import ctoaster_jobs
```

Comment:
I have tried using the import statements outside the local scopes, I was encountering error while doing that and the rest api endpoints were not working as expected. That's why the import statements are inside the local scope.

Reply:
Thank you for explaining the reason for placing the import statements inside the local scope. I understand the issues you encountered.

To address this, perhaps we can investigate why the imports are causing errors when placed at the top of the file. There might be an underlying issue that we can resolve, which will allow us to follow the best practice of keeping import statements at the top. This can help improve the readability and maintainability of our code.

Would you be able to share the specific errors you encountered when the imports were moved to the top? This might give us some clues on how to address the root cause.

Reply:
resolved.